### PR TITLE
Introduce peekaboo-util

### PR DESCRIPTION
**Pull request by Jack28**
peekaboo-util.py as a super charged replacement for scan_file.py

Add subcommand scan_file with same behaviour as scan_file.py
previously. More commands can now be easily added to the API.

### CODE REVIEW

File: bin/peekaboo-util.py [hunk 1]
```diff
@@ -43,49 +90,27 @@ def main():
...
-    result_regex = re.compile(r'.*wurde als',
-                              re.MULTILINE + re.DOTALL + re.UNICODE)
-    peekaboo = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
-    peekaboo.connect(args.socket_file)
 
-    file_snippets = []
-    for filename in args.filename:
-        file_snippets.append('{ "full_name": "%s" }' % path.abspath(filename))
-    request = '[ %s ]' % ', '.join(file_snippets)
+    scan_file_parser = subparsers.add_parser('scan_file',
```
Comment (michaelweiser) [inline bin/peekaboo-util.py]: I'd vote for 'scan-file' as command name here. Looks less techie, more like end-user interface and corresponds to peekaboo_-_util. --socket_file may be another candidate for beautification. :)

File: bin/peekaboo-util.py [hunk 2]
```diff
@@ -43,49 +90,27 @@ def main():
...
+                                             help='Scan a file and report it')
+    scan_file_parser.add_argument('-f', '--filename', action='append', required=True,
```
Comment (michaelweiser) [inline bin/peekaboo-util.py]: Just a remark as heads-up: The required option sequence is peekaboo-util.py -s &lt;sockfile&gt; scan_file -f &lt;file&gt;. When putting -s after the command name, an error is printed "error: argument -s/--socket_file is required" which can be confusing. From my point of view it's certainly consistent with the command and parser structure but dunno how easy we want to make it for users. We *could* add global arguments to each subcommand as well. No idea if it's worth the effort.

File: bin/peekaboo-util.py [hunk 3]
```diff
@@ -32,19 +32,30 @@
 import socket
 import re
 import logging
+import time
+import sys
+import inspect
+
+currentdir = path.dirname(path.abspath(inspect.getfile(inspect.currentframe())))
```
Comment (michaelweiser) [inline bin/peekaboo-util.py]: Huh? Debugging leftover or meant for production?

File: peekaboo/server.py [hunk 4]
```diff
@@ -153,6 +153,30 @@ def handle(self):
         # here we know that all samples have reported back
         self.report(submitted)
 
+    def __submit_sample__(self, api_data):
```
Comment (michaelweiser) [inline peekaboo/server.py]: That's a lot of underscores. Any particular reason?

File: bin/peekaboo-util.py (hunk 2 above)
Comment (Jack28) [reply bin/peekaboo-util.py]: The help states nicely where to put the arguments.
I will test in the next workshop how people react to it.

```
$ bin/peekaboo-util.py
usage: peekaboo-util.py [-h] [-v] [-vv] [-d] -s SOCKET_FILE {scan-file} ...
peekaboo-util.py: error: too few arguments
$ bin/peekaboo-util.py scan-file -h
usage: peekaboo-util.py scan-file [-h] -f FILENAME
```

Comment (michaelweiser) [reply bin/peekaboo-util.py]: Agreed.

**Pull request merged**